### PR TITLE
Handle no IPv4 or vsock in serve_test

### PR DESCRIPTION
**Pull request by brho**
Tested: Passed on kernels without IPv4 and without VSOCK

e.g. strace from a passing test:

socket(AF_VSOCK, SOCK_STREAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0) = -1 EAFNOSUPPORT (Address family not supported by protocol)

Signed-off-by: Barret Rhoden <brho@google.com>

### CODE REVIEW

File: cmds/cpud/serve_test.go [hunk 1]
```diff
@@ -34,14 +37,20 @@ func TestListen(t *testing.T) {
...
-		//{"vsock", "17010"},
+		{"vsock", "17010"},
 		{"unix", "@foobar"},
 	}
 
 	for _, tt := range oktests {
 		ln, err := listen(tt.network, tt.port)
 		if err != nil {
+			var sys_err *os.SyscallError
+			if errors.As(err, &sys_err) {
```
Comment (abrender) [inline cmds/cpud/serve_test.go]: nit: I think you can get rid of the switch and case with just:
```
// Ignore failures caused by lack of ipv4 or vsock.
if errors.As(err, &sys_err) && sys_err.Err == syscall.EAFNOSUPPORT {
  continue
}
```

File: cmds/cpud/serve_test.go [hunk 2]
```diff
@@ -34,14 +37,20 @@ func TestListen(t *testing.T) {
...
-		// Damn. Can't easily test this b/c environments vary so much.
-		//{"vsock", "17010"},
+		{"vsock", "17010"},
 		{"unix", "@foobar"},
 	}
 
 	for _, tt := range oktests {
 		ln, err := listen(tt.network, tt.port)
 		if err != nil {
+			var sys_err *os.SyscallError
```
Comment (Lencerf) [inline cmds/cpud/serve_test.go]: nit: use `sysErr` instead of `sys_err` to follow the convention in Go

Comment (brho) [reply cmds/cpud/serve_test.go]: done

File: cmds/cpud/serve_test.go (hunk 1 above)
Comment (brho) [reply cmds/cpud/serve_test.go]: done.  sorry i missed this earlier.  i got a github email about lencerf's comment, but not yours.

**Review by rminnich — APPROVED** [review]

**Pull request merged**